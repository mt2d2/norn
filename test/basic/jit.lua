local function maths()
  return (-10 + (2 + 2) * 20)
end

local function jit_test(m, t)
	local ret = 0

	for i=1,10000000 do
		ret = ret + m + t + maths()
	end

	return ret
end

local function float_maths()
  return 4.0 + 4.0
end

local function jit_float_test()
  return 8.4 + float_maths()
end

print(jit_test(4, 4))
print(jit_float_test())
