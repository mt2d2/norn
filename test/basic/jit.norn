def maths(): Int
	return (-10 + (2 + 2) * 20)
end

def jit_test(m: Int, t: Int): Int
	ret: Int = 0

	for i: Int = 0, i < 10000000, i=i+1 in
		ret = ret + m + t + maths()
	end

	return ret
end

def float_maths(): Float
	return 4.0 + 4.0
end

def jit_float_test(): Float
	return 8.4 + float_maths()
end

def main(): Void
	println(jit_test(4, 4))
	println(jit_float_test())
end
