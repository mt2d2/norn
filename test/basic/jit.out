780000000
16.4
